packages/sfui/frameworks/react/components/SfScrollable/SfScrollable.tsx

```diff
@@ -0,0 +1,131 @@
+      ...attributes
+    },
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    ref,
+  ) => {
+    const Tag = as || defaultScrollableTag;
+
+    const sliderOptions = useMemo(
+      () => ({
+        direction,
+        activeIndex,
+        reduceMotion,
+        drag,
+        onDragChange,
+        onScroll,
+        onPrev,
+        onNext,
+      }),
+      [direction, activeIndex, reduceMotion, drag, onDragChange, onScroll, onPrev, onNext],
+    );
+
+    const { getContainerProps, state, getNextButtonProps, getPrevButtonProps } = useScrollable(sliderOptions);
+
+    const changeDisabledClass = (isDisabled: boolean) =>
+      isDisabled ? '!ring-gray-200 !text-disabled-500' : '!ring-neutral-500 !text-neutral-500';
+    const previousButton = (...buttonClassName: Parameters<typeof classNames>) =>
+      (slotPreviousButton && <div {...getPrevButtonProps()}>{slotPreviousButton}</div>) || (
+        <SfButton
+          variant="secondary"
+          size="lg"
+          className={classNames(
+            buttonClassName,
+            changeDisabledClass(
+              typeof previousDisabled === 'boolean' ? previousDisabled : getPrevButtonProps().disabled,
+            ),
+          )}
+          square
+          slotPrefix={<SfIconChevronLeft />}
+          {...getPrevButtonProps()}
+          disabled={previousDisabled}
+        />
+      );
+
+    const nextButton = (...buttonClassName: Parameters<typeof classNames>) =>
+      (slotNextButton && <div {...getNextButtonProps()}>{slotNextButton}</div>) || (
+        <SfButton
+          variant="secondary"
+          size="lg"
+          square
+          className={classNames(
+            buttonClassName,
+            changeDisabledClass(typeof nextDisabled === 'boolean' ? nextDisabled : getNextButtonProps().disabled),
+          )}
+          slotPrefix={<SfIconChevronRight />}
+          {...getNextButtonProps()}
+          disabled={nextDisabled}
+        />
+      );
+
+    const isHorizontal = direction === SfScrollableDirection.horizontal;
+
+    return (
+      <div
+        className={classNames(
+          'items-center relative',
+          isHorizontal ? 'flex' : 'flex-col h-full inline-flex',
+          wrapperClassNames,
+        )}
+      >
+        {buttonsPlacement === SfScrollableButtonsPlacement.blocked &&
+          previousButton('!rounded-full bg-white', isHorizontal ? 'mr-4' : 'mb-4 rotate-90')}
+        <Tag
+          {...getContainerProps({
+            className: classNames(className, 'motion-safe:scroll-smooth', {
+              'overflow-x-auto flex gap-4': isHorizontal,
+              'overflow-y-auto flex flex-col gap-4': !isHorizontal,
+              'cursor-grab': state.isDragged,
+            }),
+          })}
+          {...attributes}
+        >
+          {buttonsPlacement === SfScrollableButtonsPlacement.floating &&
+            previousButton('absolute !rounded-full bg-white', isHorizontal ? 'left-4' : 'top-4 rotate-90')}
```

Comment:
Should consider implementing dedicated `z-index` value for `floating` buttons as these are rendered on the same level as list elements, 
For example there is a conflict of `absolute` positioned elements (absolute positioned `next/image`)

<img width="224" alt="image" src="https://user-images.githubusercontent.com/32792205/231111971-55798417-d225-4cf5-92bb-a6a9690a0b3d.png">

Maybe buttons would be isolated from (or stacked over) other list elements.

Reply:
What do you mean by `absolute` elements, no matter what z-index will have nth child, what matter is zindex of siblings, that how HTML works, anyhow its good suggestion to add z-index

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2571/commits/ae4261c863bfb443cd25313e377a3c6addeae14a

Reply:
In my example, image (or some its wrapper is using position `absolute`), and image just overlaps button (also positioned absolutely to the same parent element)

Reply:
@Szymon-dziewonski you implemented these changes to `block` buttons, not `floating` ones.

Reply:
![image](https://user-images.githubusercontent.com/2566152/231193587-0e235f63-41cc-4497-bfcd-b56a669ec78b.png)

Thank you - now it should be fine :) https://github.com/vuestorefront/storefront-ui/pull/2571/commits/291bc64e05da89c1e03134a79d5da51518194034